src/baseline.rs

```diff
@@ -7,22 +7,210 @@ use crate::dump::RustDocCommand;
+
+    std::fs::write(
+        &placeholder_manifest_path,
+        toml::to_string(&placeholder_manifest)?,
+    )
+    .with_context(|| "failed to write placeholder Manifest")?;
+    std::fs::write(placeholder_build_dir.join("lib.rs"), "")
+        .with_context(|| "failed to create empty lib.rs")?;
+    Ok(placeholder_manifest_path)
+}
+
+fn generate_rustdoc(
+    config: &mut GlobalConfig,
+    rustdoc: &RustDocCommand,
+    target_root: PathBuf,
+    crate_source: CrateSource,
+) -> anyhow::Result<PathBuf> {
+    let (name, version) = match crate_source {
+        CrateSource::Registry { crate_ } => (crate_.name().to_string(), crate_.version()),
+        CrateSource::ManifestPath { .. } => unimplemented!(),
+    };
+    let (build_dir, cache_dir) = match crate_source {
+        CrateSource::Registry { .. } => {
+            let crate_identifier = format!("registry-{}-{}", slugify(&name), slugify(version));
+            let cache_dir = target_root.join("cache").join(&crate_identifier);
+
+            // We assume that the generated rustdoc is untouched.
+            // Users should run cargo-clean if they experience any anomalies.
+            if cache_dir.exists() {
+                return Ok(cache_dir.join("rustdoc.json"));
+            }
+
+            let build_dir = target_root.join(crate_identifier);
+            (build_dir, cache_dir)
+        }
+        CrateSource::ManifestPath { .. } => {
+            unimplemented!()
+        }
+    };
+
+    let placeholder_manifest = create_placeholder_rustdoc_manifest(&crate_source)
+        .with_context(|| "failed to create placeholder manifest")?;
+    let placeholder_manifest_path =
+        save_placeholder_rustdoc_manifest(build_dir.as_path(), placeholder_manifest)
+            .with_context(|| "failed to save placeholder rustdoc manifest")?;
+
+    config.shell_status("Parsing", format_args!("{name} v{version} (baseline)"))?;
+    // TODO: replace (baseline) with something else
+
+    let rustdoc_path = rustdoc.dump(
+        placeholder_manifest_path.as_path(),
+        Some(&format!("{name}@{version}")),
+        false,
+    )?;
+
+    match crate_source {
+        CrateSource::Registry { .. } => {
+            // Clean up after ourselves.
+            std::fs::create_dir_all(&cache_dir).with_context(|| "failed to create cache dir")?;
+            let cache_rustdoc_path = cache_dir.join("rustdoc.json");
+            std::fs::copy(rustdoc_path, &cache_rustdoc_path)
+                .with_context(|| "failed to copy cache dir")?;
+            std::fs::remove_dir_all(build_dir).with_context(|| "failed to remove build dir")?;
+            Ok(cache_rustdoc_path)
+        }
+        CrateSource::ManifestPath { .. } => {
+            // We don't do any caching here -- since the crate is saved locally,
+            // it could be modified by the user since it was cached.
+            Ok(rustdoc_path)
```

Comment:
We can never get here or am I missing something? With the previous paths being unimplemented, this function will only work for `Registry`.

Reply:
Thanks for reviewing!
Yes, the current code only works for Registry -- this PR is a preparation for implementing the Manifest path (in the next PR I'll implement it). I've split it to make the review easier.

Reply:
Got it, so we only ever call this with the `Registry` variant?

Reply:
Yes. Sorry for not making it clear in the PR